src/core/errorHandler.js

```diff
@@ -0,0 +1,79 @@
+import React, { PropTypes } from 'react';
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+
+import { clearError, setError } from 'core/actions/errors';
+import log from 'core/logger';
+
+import 'core/css/ErrorHandler.scss';
+
+function generateHandlerId({ name = '' } = {}) {
+  return `${name}-${Math.random().toString(36).substr(2, 9)}`;
+}
+
+export class ErrorHandler {
+  constructor({ id, dispatch }) {
+    this.id = id;
+    this.dispatch = dispatch;
+  }
+
+  clear() {
+    log.info('Clearing last error for ', this.id);
+    this.dispatch(clearError(this.id));
+  }
+
+  handle(error) {
+    const info = { error, id: this.id };
+    log.info('Dispatching error action', info);
+    this.dispatch(setError(info));
+  }
+}
+
+export function withErrorHandling({
+  name, id = generateHandlerId({ name }),
+} = {}) {
+  return (WrappedComponent) => {
+    class ErrorHandlerComponent extends React.Component {
+      static propTypes = {
+        error: PropTypes.object,
+        dispatch: PropTypes.func,
+      }
+
+      render() {
+        const { error, dispatch, ...otherProps } = this.props;
+
+        const errorHandler = new ErrorHandler({ id, dispatch });
+        const props = { ...otherProps, errorHandler };
+
+        let errorInfo;
+        if (error) {
+          let errorMessages = [];
+          errorMessages = error.messages.map((msg) => <li>{msg}</li>);
+          errorInfo = (
+            <ul className="ErrorHandler-list">
+              {errorMessages}
+            </ul>
+          );
+        }
+
+        return (
+          <div>
+            {errorInfo}
+            <WrappedComponent {...props} />
```

Comment:
If you move creating the error handler to `mapDispatchToProps()` then you could remove this component and just return the `connect()` call on the wrapped component. To render the errors you could expose the `errorInfo` bit above as `ErrorList` and that could be imported instead. That would let the caller put the error messages wherever makes sense.

```jsx
import { ErrorList, withErrorHandling } from 'core/errorHandler';
class AllReviewsBase extends React.Component {
  render() {
    const { error, reviews } = this.props;
    return (
      <div className="AllReviews">
        <h1>All Reviews</h1>
        <ErrorList error={error} />
        <ul>
          {reviews.map((review) => <li>{review.text}</li>)}
        </ul>
       </div>
    );
  }
}
export default withErrorHandling({ name: "AllReviews" })(AllReviewsBase);
```

This would also avoid putting that extra `<div />` in the DOM too.